bundles/org.openhab.binding.irobot/src/main/java/org/openhab/binding/irobot/handler/RoombaHandler.java

```diff
@@ -0,0 +1,520 @@
+                    DatagramPacket identPacket = IdentProtocol.receiveResponse(identSocket);
+                    IdentProtocol.IdentData ident;
+
+                    identSocket.close();
+
+                    try {
+                        ident = IdentProtocol.decodeResponse(identPacket);
+                    } catch (JsonParseException e) {
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                                "Malformed IDENT response");
+                        return;
+                    }
+
+                    if (ident.ver < IdentData.MIN_SUPPORTED_VERSION) {
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                                "Unsupported version " + ident.ver);
+                        return;
+                    }
+
+                    if (!ident.product.equals(IdentData.PRODUCT_ROOMBA)) {
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                                "Not a Roomba: " + ident.product);
+                        return;
+                    }
+
+                    blid = ident.blid;
+                    this.blid = blid;
+                }
+
+                logger.debug("BLID is: {}", blid);
+
+                if (!config.havePassword()) {
+                    RawMQTT mqtt;
+
+                    try {
+                        mqtt = new RawMQTT(host, 8883);
+                    } catch (KeyManagementException | NoSuchAlgorithmException e1) {
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e1.toString());
+                        return; // This is internal system error, no retry
+                    }
+
+                    mqtt.requestPassword();
+                    RawMQTT.Packet response = mqtt.readPacket();
+                    mqtt.close();
+
+                    if (response != null && response.isValidPasswdPacket()) {
+                        RawMQTT.PasswdPacket passwdPacket = new RawMQTT.PasswdPacket(response);
+
+                        config.password = passwdPacket.getPassword();
+                        if (config.havePassword()) {
+                            Configuration configuration = editConfiguration();
+
+                            configuration.put(RoombaConfiguration.FIELD_PASSWORD, config.password);
+                            updateConfiguration(configuration);
+                        }
+                    }
+                }
+
+                if (!config.havePassword()) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_PENDING,
+                            "Authentication on the robot is required");
+                    scheduleReconnect();
+                    return;
+                }
+
+                logger.debug("Password is: {}", config.password);
```

Comment:
Well, it can be looked up in Thing configuration anyways.
This is not a password in traditional sense. It's not defined by the user, neither it can be changed. It is hardcoded inside the robot instead, and it is supposed to be discovered. The robot gives it away when a button on it is pressed. Official app also used this discovery algorithm. The password is considered "technical", it's not written anywhere in the manual, on some card or on the robot itself.
 But OK, your wish...

Reply:
Just asking. It's up to you.

Reply:
Meanwhile removed :)